common/src/main/java/com/revenuecat/purchases/common/offlineentitlements/OfflineCustomerInfoCalculator.kt

```diff
@@ -34,6 +44,18 @@ class OfflineCustomerInfoCalculator(
         purchasedProductsFetcher.queryPurchasedProducts(
             appUserID,
             onSuccess = { purchasedProducts ->
+                val containsAnyActiveInAppPurchase = purchasedProducts.any {
+                    it.storeTransaction.type == ProductType.INAPP && it.isActive
+                }
+                if (containsAnyActiveInAppPurchase) {
+                    val error = PurchasesError(
+                        PurchasesErrorCode.UnsupportedError,
+                        OfflineEntitlementsStrings.OFFLINE_ENTITLEMENTS_UNSUPPORTED_INAPP_PURCHASES
+                    )
+                    errorLog(COMPUTING_OFFLINE_CUSTOMER_INFO_FAILED.format(error))
+                    onError(error)
+                    return@queryPurchasedProducts
+                }
```

Comment:
I was thinking what happens if there's also an active subscription. But I guess the main use case for this is that after a successful purchase of an iapp, the app will be unconsumed and this should fail since we don't want to support consumables.

Reply:
Right, if they purchase a subscription first, they will gain access, and then, purchase a consumable, they will lose access until the backend is back up. I think the cases of this happening in a short time frame will be small though.